### ISSUE TITLE
Test set data seen during training

### ISSUE THREAD

**Issue by adrianastan** [Jiaxin-Ye/TIM-Net_SER#16]
Hi,

I just noticed that in your code you are presenting the test set data to the [model.fit() method](https://github.com/Jiaxin-Ye/TIM-Net_SER/blob/8bd7f6f4813150ca2adf7c1bce129c5f9ba81940/Code/Model.py#L112) and select the best model based on it, instead of using a separate validation set or early stopping/fixed number of epochs. 

However, this means that your reported results on the test set are "cherry picked". Do you have a report on the performance of your model when the test data is not used for validation during training? The same method seems to have been applied to your previous papers on SER.

Thanks,
Adriana

**Comment by Jiaxin-Ye** [comment]
Dear Dr. Stan,

Thank you for your interest in this project and your valuable suggestions. I sincerely apologize for any inconvenience caused to you. Upon carefully rechecking the code, I have finished the experimental supplementary results and will update them on GitHub and arXiv in the next few days. 

These are our supplementary results under 10-fold cross-validation (UAR % / WAR %), and all models are trained with the original hyperparameters for 500 epochs, which are evaluated only in the last epoch to avoid overfitting issues. The updated README file provides further details on the experiment specifics. Feel free to refer to it for more information.

<!DOCTYPE html><p></p><figure class="md-table-fig" cid="n86" mdtype="table" style="box-sizing: border-box; margin: 1.2em 0px; overflow-x: auto; max-width: calc(100% + 16px); padding: 0px; cursor: default;">

UAR (%)/WAR (%) | CASIA | EMODB | EMOVO | IEMOCAP | RAVDESS | SAVEE
-- | -- | -- | -- | -- | -- | --
TIM-Net | 91.08 / 91.08 | 89.19 / 90.28 | 86.56 / 86.56 | 69.00 / 68.29 | 90.04 / 90.07 | 77.26 / 79.36

</figure><blockquote cid="n111" mdtype="blockquote" style="box-sizing: border-box; margin: 0.8em 0px; border-left-width: 4px; border-left-style: solid; border-left-color: rgb(223, 226, 229); padding: 0px 15px; color: rgb(119, 119, 119);"><p></p></blockquote>

I greatly appreciate your feedback, which helps us continually refine and enhance the quality of our work. If you have any further questions or feedback, please don't hesitate to reach out to me. Thank you very much for your understanding and support.

Best regards,

Jiaxin Ye
